backend/LexBoxApi/Services/EmailService.cs

```diff
@@ -31,15 +32,39 @@ public class EmailService
 
     public async Task SendForgotPasswordEmail(string jwt, User user)
     {
-        var message = new MimeMessage();
-        message.To.Add(new MailboxAddress(user.Name, user.Email));
+        var email = StartUserEmail(user);
         var httpContext = _httpContextAccessor.HttpContext;
         ArgumentNullException.ThrowIfNull(httpContext);
         // returnTo is a svelte app url
         var forgotLink = _linkGenerator.GetUriByAction(httpContext, "LoginRedirect", "Login", new { jwt, returnTo = "/resetPassword" });
         ArgumentException.ThrowIfNullOrEmpty(forgotLink);
-        await RenderEmail(message, new ForgotPasswordEmail(user.Name, forgotLink));
-        await SendEmailAsync(message);
+        await RenderEmail(email, new ForgotPasswordEmail(user.Name, forgotLink));
+        await SendEmailAsync(email);
+    }
+
+    /// <param name="newEmail">If the user is trying to change their address, this is the new one, otherwise null.</param>
+    public async Task SendVerifyAddressEmail(string jwt, User user, string? newEmail = null)
+    {
+        var email = StartUserEmail(user, newEmail);
+        var httpContext = _httpContextAccessor.HttpContext;
+        ArgumentNullException.ThrowIfNull(httpContext);
+        var queryParam = string.IsNullOrEmpty(newEmail) ? "verifiedEmail" : "changedEmail";
```

Comment:
I don't love the magic strings here, but I don't really have a good suggestion of what to do instead. Maybe if we used a consistent unique queryParam name with a value indicating what was done, then you could lookup the unique name to figure out what uses it, `verifiedEmail` and `changedEmail` are kinda hard to search for across the code base.

Reply:
It would be nice to share a constant here between the back and front ends. E.g. something like this:
https://typegen.readthedocs.io/en/latest/overview.html

Reply:
That's an interesting idea, I'd never thought of it. It does however complicate our build process even more which I don't like.